File: programmers/난이도별/level01.하샤드_수/JiaLee0707.c
#include <stdio.h>
#include <stdbool.h>
#include <stdlib.h>

bool solution(int x) {
    bool answer = true;

    int xx = x;
    int sum = 0;

    while (xx != 0) {
        sum += xx % 10;
        xx /= 10;
    }

    if (x % sum != 0) answer = false;

    return answer;
}

File: programmers/난이도별/level01.삼총사/JiaLee0707.js
const threeMusketeers = (members, candidates) => {
    let answer = 0;
    const membersLength = members.length;
    
    if (membersLength > 2) {
        const sum = members.reduce((acc, cur) => acc + cur, 0);
        return sum === 0;
    }
    
    candidates.forEach((newMember, index) => {
        answer += threeMusketeers([...members, newMember], candidates.slice(0, index));
    });
    
    return answer;
}

function solution(number) {
    return threeMusketeers([], number);
}
